feat(blamer-kill-ring-commit-hash): copy line's commit to kill-ring

Boruch-Baum opened the pull request:
+ resolves #56

Artawower commented:
Thank you for your contribution! Could you please fix some errors in ci? https://github.com/Artawower/blamer.el/actions/runs/7682081509/job/20936706320?pr=57

Also, I think we can make this function ;;;###autoload

Boruch-Baum commented:
I've examing the CI log. On the one hand, it seems that the errors
indicate a bug in the CI software, not in my PR code. On the other hand,
the CI software did its job well in uncovering a needless call and
needless variable definition, so I made a change that might satisfy the
CI software.

The CI software seems to not recognize the function definition
'lexical-let' which Emacs describe-function claims was "Probably
introduced at or before Emacs version 24.3". Because of that, the CI
software proceeds to misinterpret the local variable definitions
(this-line-number and current-buffer) as function calls instead of local
variable definitions.

The lexical scope was once-upon-a-time required because current-buffer
was once-upon-a-time used in the callback lambda function. Let's see
whether these changes help.

I also added the autoload.


On 2024-01-27 17:20, Art wrote:
>    Thank you for your contribution! Could you please fix some errors in
>    ci?
>    [1]https://github.com/Artawower/blamer.el/actions/runs/7682081509/job/2
>    0936706320?pr=57
>
>    Also, I think we can make this function ;;;###autoload
>
>    —
>    Reply to this email directly, [2]view it on GitHub, or [3]unsubscribe.
>    You are receiving this because you authored the thread. Message ID:
>    ***@***.***>
>
> References
>
>    1. https://github.com/Artawower/blamer.el/actions/runs/7682081509/job/20936706320?pr=57
>    2. https://github.com/Artawower/blamer.el/pull/57#issuecomment-1913398530
>    3. https://github.com/notifications/unsubscribe-auth/AAOE3KDA42ISTE3GM7WS3L3YQWRUTAVCNFSM6AAAAABCNXGO2KVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJTGM4TQNJTGA

--
hkp://keys.gnupg.net
CA45 09B5 5351 7C11 A9D1  7286 0036 9E45 1595 8BC0

Artawower commented:
thank you for your understanding 🫡
